ls1intum/Artemis
# `Assessment`: Give tutors hints which elements they missed when assessing example submission

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


- When assessing a modeling example submissions, it is very hard for tutors to now which elements they missed assessing. I now display this by highlighting them in light-blue
- When assessing a text example submissions, it is very hard for tutors to now text blocks they have to assess. I now display this by highlighting them in light-blue + tutors can now only select those that they should assess.
### Description


- Send the reference ids of model elements / text blocks that should be assessed to the client. Then just highlight these in lightblue :) 

### Steps for Testing
- Create a modeling exercise / text exercise with an example submission that the tutors need to assess.
- Assess this example submission as a tutor and miss some elements -> Confirm that they are displayed in light blue until you assessed them all correctly! 

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
- Added server test to confirm only the reference ids are send of th example solution and no other data.

### Screenshots



Before:

https://user-images.githubusercontent.com/29718932/194674161-ea5716ae-827d-427d-ba36-065874d40aa5.mp4


https://user-images.githubusercontent.com/29718932/195399871-a8b26ffa-8879-44e8-99a1-72f3104780e0.mp4


After:


https://user-images.githubusercontent.com/29718932/194674302-5de0194c-0e0c-450e-a75e-4e838bea1407.mp4


https://user-images.githubusercontent.com/29718932/195375707-079fde3f-bb64-4bb5-9f17-3b1aa88d2edc.mp4
